File: Python/save_manager.py
# ========================================
# 坦克大战 - 存档管理
# ========================================

import os
import json

SAVE_DIR = "PlayerData"
SAVE_FILE = "LevelSaves.dat"
SAVE_PATH = os.path.join(SAVE_DIR, SAVE_FILE)


def ensure_save_dir():
    """确保存档目录存在"""
    if not os.path.exists(SAVE_DIR):
        os.makedirs(SAVE_DIR)


def save_progress(level):
    """保存关卡进度"""
    ensure_save_dir()
    try:
        with open(SAVE_PATH, 'w') as f:
            json.dump({'level': level}, f)
        return True
    except Exception as e:
        print(f"保存失败: {e}")
        return False


def load_progress():
    """读取关卡进度，返回关卡编号，如果没有存档返回 1"""
    ensure_save_dir()
    if not os.path.exists(SAVE_PATH):
        return 1  # 没有存档则从第1关开始
    try:
        with open(SAVE_PATH, 'r') as f:
            data = json.load(f)
            return data.get('level', 1)
    except Exception:
        return 1


def has_save():
    """检查是否有存档"""
    return os.path.exists(SAVE_PATH)

File: C++/src/SaveManager.cpp
#include "SaveManager.h"
#include <fstream>
#include <iostream>
#include <sys/stat.h>

#ifdef _WIN32
#include <direct.h>
#define MKDIR(path) _mkdir(path)
#else
#include <sys/stat.h>
#define MKDIR(path) mkdir(path, 0755)
#endif

const std::string SaveManager::SAVE_DIR = "PlayerData";
const std::string SaveManager::SAVE_FILE = "LevelSaves.dat";

void SaveManager::ensureSaveDir() {
    struct stat st;
    if (stat(SAVE_DIR.c_str(), &st) != 0) {
        // 目录不存在，创建
        if (MKDIR(SAVE_DIR.c_str()) != 0) {
            std::cerr << "创建存档目录失败: " << SAVE_DIR << std::endl;
        }
    }
}

std::string SaveManager::getSavePath() {
    return SAVE_DIR + "/" + SAVE_FILE;
}

void SaveManager::saveProgress(int level) {
    ensureSaveDir();
    std::string path = getSavePath();
    
    std::ofstream file(path, std::ios::binary);
    if (file.is_open()) {
        file.write(reinterpret_cast<const char*>(&level), sizeof(level));
        file.close();
    } else {
        std::cerr << "保存失败: " << path << std::endl;
    }
}

int SaveManager::loadProgress() {
    ensureSaveDir();
    std::string path = getSavePath();
    
    std::ifstream file(path, std::ios::binary);
    if (file.is_open()) {
        int level = 1;
        file.read(reinterpret_cast<char*>(&level), sizeof(level));
        file.close();
        return level;
    }
    return 1;  // 没有存档，从第1关开始
}

bool SaveManager::hasSave() {
    std::string path = getSavePath();
    std::ifstream file(path);
    return file.good();
}